client/dataEntry.js

```diff
@@ -31,20 +31,12 @@ Template.dataEntry.helpers({
   },
 
   isSelectedRoundClosed: function() {
-    if(!this.roundId) {
-      // If there's no round selected, then the selected round is definitely
-      // *not* closed =)
-      return false;
-    }
-    return Rounds.findOne(this.roundId).isClosed();
+    round = Rounds.findOne(this.roundId);
```

Comment:
How about this variant:

    return Rounds.findOne(this.roundId).isClosed();

This will throw both for an invalid and non existent roundId. I think that makes sense. These functions aren't intended to work without a roundId.

***
This would sidestep the philosophical issue, but here are my thoughts anyway: 

Yeah, in the real world, DBs do become corrupted occasionally, for several reasons. This is probably especially true for a DB lacking transactions and foreign key constraints.

The way I like to handle this is with a "sanity checker". Code that checks the entire DB for inconsistencies. Like that all `roundId` fields must contain a valid id for an existing Round, and so on. You can even make this self maintaining with some wizardry.

It typically runs every night and sends out an email to the team with the result. Normally you fix any corruption the same day. Less ambitiously, you can have a sanity checker page for manual checking.

With this system I have a high level of confidence that the DB is consistent, and I feel good about not checking things in the regular app code. Then again that is a calculated risk, and I understand that it's not inherently wrong to pick another tradeoff.


I might actually look into writing a generic Sanity Checker package for Mongo and/or Meteor.